src/MicroserviceErrorHandler.js

```diff
@@ -15,19 +15,21 @@ module.exports = class MicroserviceErrorHandler {
 
     extractServiceName(options) {
         const serviceName = options.serviceName || (options.microservice || {}).name;
-        if(!serviceName) {
-            const msg = 'Configuration value for @joinbox/loopback-microservice#errorHandler' +
-                ' is missing: Please add the property "serviceName" to the "params" section of the' +
-                ' middleware config or copy your general microservice config (containing the name)'
-            ' by adding "microservice": "${microservice}"';
-            throw new MicroserviceError(msg);
+        if (serviceName) {
+            return serviceName;
         }
-        return serviceName;
+
+        const msg = 'Configuration value for @joinbox/loopback-microservice#errorHandler' +
+            ' is missing: Please add the property "serviceName" to the "params" section of the' +
+            ' middleware config or copy your general microservice config (containing the name)' +
+            // eslint-disable-next-line no-template-curly-in-string
+            ' by adding "microservice": "${microservice}"';
+        throw new MicroserviceError(msg);
     }
 
-    _copyAndAddSafeFields(options, fields) {
+    _copyAndAddSafeFields(options, fields = []) {
```

Comment:
Whats the intention of copying the options?

Reply:
We get this options by reference (usually via configuration). I don't know where they come from and where they are used outside of the error handler. Modifying them by reference might lead to side-effects.

Reply:
Btw: I just saw that i still modify them wrongly (below when i push into the array), will fix that.